### DISCUSSION TITLE
Distinguish RP2040 / RP2350 chip in sketch?

### DISCUSSION THREAD

**Discussion by FeuerSturm** [earlephilhower/arduino-pico#2367] [Q&A] [answered]
Hi,

I'm probably blind and the answer is really easy, but I just cannot see it:
**How can I distinguish (within a sketch) if a _RP2040_ or _RP2350_ chip based board was selected to compile a sketch for?**

Checking for ARDUINO_ARCH_RP2040 can be both chips, how can I check further?

Thanks in advance!

**Comment by earlephilhower** [comment, marked as answer]
````
#if defined(PICO_RP2350)
....
#else if defined(PICO_RP2040)
....
#2189

**Reply by FeuerSturm** [reply]
Thank you very much, I knew I was just blind! 👍 

You sir are my personal hero! 🥇
